packages/nextjs/hooks/scaffold-eth/useSelectedNetwork.ts

```diff
@@ -0,0 +1,8 @@
+import scaffoldConfig from "~~/scaffold.config";
+import { useGlobalState } from "~~/services/store/store";
+import { AllowedChainIds } from "~~/utils/scaffold-eth";
+
+export function useSelectedNetwork(chainId?: AllowedChainIds) {
+  const targetNetwork = useGlobalState(({ targetNetwork }) => targetNetwork);
+  return scaffoldConfig.targetNetworks.find(targetNetwork => targetNetwork.id === chainId) ?? targetNetwork;
+}
```

Comment:
Not for this PR, but in future we could improve this. 

I think there is similarity b/w `useTargetNetwork` and `useSelectedNetwork`. And feels like we could get the functionality of this hook inside `useTargetNetwork` if it would have accepted `chainId` as param. 

Don't change anything but something we could tackle in future.